frontend/src/components/Rides/RequestRide.js

```diff
@@ -32,7 +32,7 @@ class RequestRide extends Component {
             this.setState({openConfirmation: true});
         }
         else {
-            this.setState(initialState);
+            this.state = initialState;
```

Comment:
I think this is important to note from the React docs:
"Never mutate this.state directly, as calling setState() afterwards may replace the mutation you made. Treat this.state as if it were immutable."

https://reactjs.org/docs/react-component.html

Reply:
> not sure why do you need to this.state = initialState here if state is defined.

Here it is for state undefined. Otherwise, the code won't work. 
I'm aware of the problem Alex mentions, but this line is for initializing this.state, and it won't be executed once it's initialized (or outside of the constructor). I'm doing this to avoid having `const initialState` at two different places (both the constructor and the reset function). Since the initialState still contains hardcoded information for testing, it will be annoying to modify the both while testing. But once we have the actual userid info, I'll do it the normal way, which only use setState outside of the constructor.